modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/openapi/contributor/ExtensionsOpenAPIContributor.java

```diff
@@ -0,0 +1,186 @@
+
+	@Override
+	public void contribute(OpenAPI openAPI, OpenAPIContext openAPIContext)
+		throws Exception {
+
+		Map<String, Schema> schemas = openAPI.getComponents(
+		).getSchemas();
+
+		if (schemas.isEmpty()) {
+			return;
+		}
+
+		Map<String, EntityField> entityFieldsMap = getEntityModelFromJaxrs(
+			openAPIContext.getPath());
+
+		if ((entityFieldsMap == null) || entityFieldsMap.isEmpty()) {
+			return;
+		}
+
+		for (Map.Entry<String, Schema> schemaEntry : schemas.entrySet()) {
+			Schema schema = schemaEntry.getValue();
+
+			if (schema.getProperties() != null) {
+				_processSchemaProperties(schema, entityFieldsMap);
+			}
+		}
+	}
+
+	@Activate
+	protected void activate(BundleContext bundleContext)
+		throws InvalidSyntaxException {
+
+		_bundleContext = bundleContext;
+
+		_serviceTrackerMap = ServiceTrackerMapFactory.openMultiValueMap(
+			bundleContext, null, "osgi.jaxrs.application.select");
+	}
+
+	@Deactivate
+	protected void deactivate() {
+		_serviceTrackerMap.close();
+	}
+
+	protected Map<String, EntityField> getEntityModelFromJaxrs(String path)
+		throws Exception {
+
+		String applicationName = _findApplicationName(path);
+
+		if (applicationName != null) {
+			String applicationSelectFilter =
+				"(osgi.jaxrs.name=" + applicationName + ")";
+
+			Object resourcesObject = _serviceTrackerMap.getService(
+				applicationSelectFilter);
+
+			if (resourcesObject instanceof List<?>) {
+				List<?> resources = (List<?>)resourcesObject;
+
+				for (Object resource : resources) {
+					if (resource instanceof EntityModelResource) {
+						EntityModelResource entityModelResource =
+							(EntityModelResource)resource;
+
+						EntityModel entityModel =
+							entityModelResource.getEntityModel(
+								new MultivaluedHashMap<>());
+
+						if (entityModel != null) {
+							return entityModel.getEntityFieldsMap();
+						}
+					}
+				}
+			}
+		}
+
+		return null;
+	}
+
+	private String _findApplicationName(String path) {
+		try {
+			String applicationBase = StringUtil.removeFirst(path, "/o");
+
+			applicationBase = StringUtil.replaceLast(applicationBase, '/', "");
+
+			String filter =
+				"(osgi.jaxrs.application.base=" + applicationBase + ")";
+
+			Collection<ServiceReference<Application>> serviceReferences =
+				_bundleContext.getServiceReferences(Application.class, filter);
+
+			if ((serviceReferences != null) && !serviceReferences.isEmpty()) {
+				for (ServiceReference<Application> serviceReference :
+						serviceReferences) {
+
+					return (String)serviceReference.getProperty(
+						"osgi.jaxrs.name");
+				}
+			}
+		}
+		catch (InvalidSyntaxException invalidSyntaxException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(invalidSyntaxException);
+			}
+
+			return null;
+		}
+
+		return null;
+	}
+
+	private void _processSchemaProperties(
+		Schema schema, Map<String, EntityField> entityFieldsMap) {
+
+		Map<?, ?> properties = schema.getProperties();
```

Comment:
Why do you use ? instead of the real type?

Reply:
I'm not sure xD Will fill in the correct types